src/app/loadout/loadout-ui/menu-hooks.tsx

```diff
@@ -225,19 +207,45 @@ export function searchAndSortLoadoutsByQuery(
   language: DimLanguage,
   loadoutSort: LoadoutSort
 ) {
-  const loadoutQueryPlain = plainString(query, language);
-  return _.sortBy(
-    loadouts.filter(
-      (loadout) =>
-        !query ||
-        plainString(loadout.name, language).includes(loadoutQueryPlain) ||
-        (!isInGameLoadout(loadout) &&
-          loadout.notes &&
-          plainString(loadout.notes, language).includes(loadoutQueryPlain))
-    ),
-    (l) => (isInGameLoadout(l) ? 0 : 1),
+  const sortCollator = new Intl.Collator(language, {
+    numeric: true,
+    usage: 'sort',
+    sensitivity: 'accent',
+  });
+  const filterCollator = new Intl.Collator(language, { usage: 'search', sensitivity: 'base' });
+
+  const normalizedQuery = query.normalize('NFC');
+
+  // Unfortunately Collator does not have a substring search method
+  // See https://github.com/adobe/react-spectrum/blob/7f63e933e61f20891b4cf3f447ab817f918cb263/packages/%40react-aria/i18n/src/useFilter.ts#L58-L76
+  const contains = (string: string) => {
+    if (normalizedQuery.length === 0) {
+      return true;
+    }
+
+    string = string.normalize('NFC');
+
+    let scan = 0;
+    const sliceLen = normalizedQuery.length;
+    for (; scan + sliceLen <= string.length; scan++) {
+      const slice = string.slice(scan, scan + sliceLen);
+      if (filterCollator.compare(normalizedQuery, slice) === 0) {
+        return true;
+      }
+    }
```

Comment:
Not 100% sure if that's correct Unicode-aware handling, since this will happily pull apart UTF-16 surrogate pairs.